acme/src/main/java/io/micronaut/acme/services/AcmeService.java

```diff
@@ -278,7 +303,7 @@ private boolean writeCombinedFile(Certificate certificate) {
                     try (BufferedWriter writer = Files.newBufferedWriter(domainCsr.toPath(), WRITE, CREATE, TRUNCATE_EXISTING)) {
                         certificate.writeCertificate(writer);
                     }
-                    eventPublisher.publishEvent(new CertificateEvent(getCurrentCertificate(), domainKeyPair, false));
+                    eventPublisher.publishEvent(new CertificateEvent(domainKeyPair, false, getFullCertificateChain()));
```

Comment:
if getFullCertificateChain returns an empty array, we should publish a the event. Since `CertificateEvent`  assumes a non empty array due to `CertificateEvent::getCert` where it gets the first certificate in the array. should we?

Reply:
Resolved via https://github.com/micronaut-projects/micronaut-acme/pull/161/commits/6a5dd4466aadfa3ee97dec776869d8940a036cdc